### ISSUE TITLE
Clippy false positive on Deserializer lifetimes

### ISSUE THREAD

**Issue by dtolnay** [serde-rs/serde#854]
The false positive in https://github.com/Manishearth/rust-clippy/issues/292 is going to hit users really hard in the next release.

```rust
fn f<'de, D: Deserializer<'de>>(deserializer: D) -> Result<F, D::Error> {
    unimplemented!()
}
```

```
warning: explicit lifetimes given in parameter types where they could be elided
 --> src/main.rs:7:1
  |
7 |   fn f<'de, D: Deserializer<'de>>(deserializer: D) -> Result<F, D::Error> {
  |  _^ starting here...
8 | |     unimplemented!()
9 | | }
  | |_^ ...ending here
  |
  = note: #[warn(needless_lifetimes)] on by default
```

@oli-obk any sense of how difficult this would be to fix?

**Comment by oli-obk** [comment]
No, but I'll make it a priority

**Comment by oli-obk** [comment]
Should be fixed now in clippy 0.0.123.

**Comment by dtolnay** [comment]
Thank you so much!
